api/desecapi/models.py

```diff
@@ -227,3 +230,68 @@ def save(self, *args, **kwargs):
 
     class Meta:
         ordering = ('created',)
+
+
+class RRset(models.Model):
+    # TODO Do these two fields really make sense? Meaning is limited when deleting + recreating an RRset
+    created = models.DateTimeField(auto_now_add=True)
+    updated = models.DateTimeField(null=True)
+    domain = models.ForeignKey(Domain, on_delete=models.CASCADE, related_name='rrsets')
+    subname = models.CharField(max_length=178, blank=True)
+    type = models.CharField(max_length=10)
+    records = models.CharField(max_length=64000, blank=True)
+    ttl = models.PositiveIntegerField(validators=[MinValueValidator(1)])
+    _dirty = False
+
+    class Meta:
+        unique_together = (("domain","subname","type"),)
+
+    def __setattr__(self, attrname, val):
+        setter_func = 'setter_' + attrname
+        if attrname in self.__dict__ and callable(getattr(self, setter_func, None)):
+            super().__setattr__(attrname, getattr(self, setter_func)(val))
+        else:
+            super().__setattr__(attrname, val)
+
+    def setter_records(self, val):
+        if val != self.records:
+            self._dirty = True
+
+        return val
+
+    def setter_ttl(self, val):
+        if val != self.ttl:
+            self._dirty = True
+
+        return val
+
+    def update_pdns(self):
+        from .serializers import RRsetSerializer
```

Comment:
I think the problem lies deeper within the organization of the JSON parsing and passing. You use `RRsetSerializer` to obtain data passed to `pdns.set_rrsets`, hence the import. The serializer here is needed as the data is stored serialized in the database -- a choice that I criticized in another comment on this PR. While I can see the point of doing this, the import issue here is a consequence of that: if the data was "properly" (in the sense of my other comment) stored, we didn't need any serializer here.

If you insist that storing serialized JSON is the way to go here, then I think `pdns.py` should provide an appropriate interface to deal with this; rather than the model deserializing the DB-stored values. (As you also brought up the pdns API as one reason to store the entire rrset serialized.)

That being said, I propose to store records not serialized, as this will save us the `jq` usage, the serializer import and parsing across the project on different layers (that were not intended to deal with (de)serialization).